nupic/algorithms/KNNClassifier.capnp

```diff
@@ -0,0 +1,41 @@
+@0x99e493bfb8fbad94;
+
+# Next ID: ???
+struct KNNClassifierProto {
+  version @0 :UInt16;
+  verbosity @1 :UInt8;
+
+  cellsPerCol @2 :UInt32;
+  k @3 :UInt32;
+  distanceNorm @4 :Float32;
+  distanceMethod @5 :Text;
+  distanceThreshold @6 :UInt32;
+  doBinarization @7 :Bool;
+  binarizationThreshold @8 :Float32;
+  useSparseMemory @9 :Bool;
+  sparseThreshold @10 :Float32;
+  relativeThreshold @11 :Bool;
+  numWinners @12 :UInt32;
+  numSVDSamples @13 :Int32;
+  numSVDDims @14 :Int32;
+  fractionOfMax @15 :Float32;
+  replaceDuplicates @16 :Bool;
+  maxStoredPatterns @17 :Int32;
+
+  iterationIdx @18 :Int32;
+
+  numPatterns @19 :Int32;
```

Comment:
Because `numPatterns` can be `None` which is represented by `-1` (which mismatch `UInt32`)